synapse/lib/hiveauth.py

```diff
@@ -993,18 +1036,20 @@ def getAllowedReason(self, perm, gateiden=None, default=False):
 
                 for allow, path in info.get('rules', ()):
                     if perm[:len(path)] == path:
-                        return (allow, f'Matched role rule ({textFromRule((allow, path))}) for role {role.name} on gate {gateiden}.')
+                        return (allow, {'rule': path, 'isgate': True, 'isrole': True, 'rolename': role.name,
```

Comment:
Seems like we could just check if `rolename`/`gateiden` have a value rather than including the additional `isgate`/`isrole` bool values